### PR TITLE
Fix netlify PR preview

### PR DESCRIPTION
**Pull request by FynnBe**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@ bioimageio.core
 lxml
 typer
 boltons
+git
```
Comment (oeway) [inline requirements.txt:6]: Not sure this will work, git is not a pypi package (it's a conda package though).

**Comment by oeway** [comment]
Since the git clone with depth 1 is equivalent to downloading the zipped archive of gh-pages. I would skip the git command but use python to download the zip file. It's always good to avoid system command to achieve better portability.

File: requirements.txt (hunk 1 above)
Comment (FynnBe) [reply requirements.txt:6]: yeah, it doesn't 😢

**Comment by FynnBe** [comment]
> Since the git clone with depth 1 is equivalent to downloading the zipped archive of gh-pages. I would skip the git command but use python to download the zip file. It's always good to avoid system command to achieve better portability.

Using git in the script is nice as you can set local branches where you want that the script is then using...  

is there another simple way to have git available during netlify build?

**Comment by oeway** [comment]
> Using git in the script is nice as you can set local branches where you want that the script is then using...

The downside is that you make the script less deterministic (e.g. git command can fail depending on the current git status). IMO, when we only need to download some files, using a normal http is more robust and reproducible.

> is there another simple way to have git available during netlify build?

Not as I know.

**Comment by FynnBe** [comment]
> The downside is that you make the script less deterministic (e.g. git command can fail depending on the current git status). IMO, when we only need to download some files, using a normal http is more robust and reproducible.

downloading files from [raw.githubuser](https://raw.githubusercontent.com/) can have a delay. especially when running the CI multiple times the content hosted at raw. will be outdated for a short time. This makes the script indeterministic. if the "git status" breaks the workflow this is desired as we use the gh-pages branch for deployment. if that one is not available or in a unintended state it is totally fine for the CI to fail because of it. Besides, if the gh-pages branch is in a weird state the download of files hosted there would fail as well.

An alternative to git is an SVN client: https://docs.github.com/en/get-started/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients

**Comment by oeway** [comment]
> > The downside is that you make the script less deterministic (e.g. git command can fail depending on the current git status). IMO, when we only need to download some files, using a normal http is more robust and reproducible.
> 
> downloading files from [raw.githubuser](https://raw.githubusercontent.com/) can have a delay. especially when running the CI multiple times the content hosted at raw. will be outdated for a short time. This makes the script indeterministic. if the "git status" breaks the workflow this is desired as we use the gh-pages branch for deployment. if that one is not available or in a unintended state it is totally fine for the CI to fail because of it. Besides, if the gh-pages branch is in a weird state the download of files hosted there would fail as well.
> 
> An alternative to git is an SVN client: https://docs.github.com/en/get-started/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients

I wouldn't count a delay as indeterministic, that's different thing for me. It's undeniable fact that git is much more complex than http, thus more error prone. And for the netlify case, seems we don't have the choice too.

For the delay, it seems you can by pass the cache too: https://stackoverflow.com/a/67465908

**Comment by FynnBe** [comment]
> For the delay, it seems you can by pass the cache too: https://stackoverflow.com/a/67465908

seems to be only applicable to gists... ?

**Comment by oeway** [comment]
> > For the delay, it seems you can by pass the cache too: https://stackoverflow.com/a/67465908
> 
> seems to be only applicable to gists... ?

Yeah, noticed, sorry, see the updated comment.

**Comment by FynnBe** [comment]
> I wouldn't count a delay as indeterministic

well, at the time the script runs it is unknown which file version is used and there is not even a way of logging which version was given (other than comparing file hashes).

> BTW, are you sure there is a delay? From what I understand now, it's normal that for the browser to cache raw git content, because the http header instruct the browser to cache Cache-Control: max-age=300, but for a python requests, it doesn't really do caching, so I would expect that to be an non issue.

not sure

Anyway, we need to download a folder from the gh-pages branch though, not just known individual files.. thus my mentioning of svn.
Afaik, there is no single http request that could be used to obtain the gh-pages/rdfs folder for example

**Comment by oeway** [comment]
> 


You can get the entire branch by one call: https://github.com/bioimage-io/collection-bioimage-io/archive/refs/heads/gh-pages.zip

**Comment by oeway** [comment]
> > 
> 
> You can get the entire branch by one call: https://github.com/bioimage-io/collection-bioimage-io/archive/refs/heads/gh-pages.zip

If you want to make sure it give the latest branch, you can 1) do a github api request to determine the latest hash 2) construct an absolute URL like this: https://github.com/bioimage-io/collection-bioimage-io/archive/b1373af7cc7acb6e906fc4eb8533f87026ad5415.zip

This I believe will by pass any caching, since you have the commit hash in the URL.

**Comment by FynnBe** [comment]
awesome. I suppose that should work!

**Pull request merged**